app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class Users::SessionsController < Devise::SessionsController
+  def new
+    session.delete(:mfa)
+    super
+  end
+
+  def create
+    self.resource = warden.authenticate!(auth_options)
+
+    if multi_factor_enabled?
+      # preserve the stored location
+      stored_location = stored_location_for(resource)
+
+      # log out the user (this will also clear stored location)
+      warden.logout
```

Comment:
Question: Is this the line that keeps a user from being able to bypass MFA via manual URL entry?